public_html/wp-content/mu-plugins/blocks/assets/src/shared/featured-image/index.js

```diff
@@ -0,0 +1,186 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies.
+ */
+const { Component } = wp.element;
+
+/**
+ * Displays featured image, can be linked with block control for size. Fetches the smallest possible image size required to render the current size.
+ */
+export default class FeaturedImage extends Component {
+
+	/**
+	 * Constructor for component. Size can be provided, if height and width both are provided, then they take precedence over size.
+	 *
+	 * @param props Props for function.
+	 * @param props.wpMediaDetails Available sizes of images in the format as returned by WP API. This is the `sizes` object inside `media_details` inside `wp:featuredMedia` object. Not required if `props.availableSizes is present.
+	 * @param props.availableSizes Available sizes of image along with source URL. Not required if `props.wpMediaDetails` is present. Should be in this format:
+	 * [
+	 * 		{
+	 * 		 	height: 999,
+	 * 		 	width: 999,
+	 * 		 	source_url: https://link.to/image-size-1.png
+	 * 		},
+	 * 		{
+	 * 			height: 500,
+	 * 			width: 500,
+	 * 			source_url: https://link.to/image-size-2.png
+	 * 		}
+	 * 		...
+	 * ]
+	 * @param props.height Height in pixels for image.
+	 * @param props.width Width in pixels for image.
+	 * @param props.className Classname for image element
+	 * @param props.alt Alt text for image
+	 * @param props.onChange Function callback for when Image URL is changed. Current URL will be passed as a parameter.
+	 */
+	constructor( props ) {
+		super( props );
+	}
```

Comment:
@vedanshujain , why is the `constructor()` needed when it's not doing anything? Is it just there to document the props? Would that be better done w/ `propTypes` ?

Reply:
This constructor is not empty anymore and initializes `state`. Resolving this conversation.

Reply:
Thanks!